src/color_wheel.cpp

```diff
@@ -571,4 +667,33 @@ void ColorWheel::dropEvent(QDropEvent* event)
     }
 }
 
+void ColorWheel::clearHarmonies()
+{
+    p->ring_editors.clear();
+    p->current_ring_editor = -1;
+    Q_EMIT harmonyChanged();
+    update();
+}
+
+void ColorWheel::addHarmony(double hue_diff, bool editable, int symmetric_to, int opposite_to)
+{
+    hue_diff = normalize(hue_diff);
+    if (symmetric_to >= 0 && opposite_to >= 0)
+        throw std::runtime_error("incorrect call to addHarmony: harmony cannot be both symmetric and opposite");
```

Comment:
Good idea. Also the correct way to add connected harmonies should be documented.